### PR TITLE
modify laplace2d to hydra style

### PR DESCRIPTION
**Pull request by xusuyong**
### PR types

Others
### PR changes

Others 
### Describe

laplace2d 案例代码重构

### CODE REVIEW

File: docs/zh/examples/laplace2d.md [hunk 1]
```diff
@@ -2,6 +2,18 @@
...
 
+=== "模型训练命令"
+
+    ``` sh
+    # linux
+    wget https://paddle-org.bj.bcebos.com/paddlescience/datasets/laplace2d/laplace2d_dataset.tar
+    # windows
+    # curl https://paddle-org.bj.bcebos.com/paddlescience/datasets/laplace2d/laplace2d_dataset.tar --output laplace2d_dataset.tar
+    # unzip it
+    tar -xvf laplace2d_dataset.tar
```
Comment (HydrogenSulfate) [inline docs/zh/examples/laplace2d.md:13]: 这个模型没有来自文件的数据集，可以删除除最后一行外的代码

File: examples/laplace/laplace2d.py [hunk 2]
```diff
@@ -50,11 +47,11 @@ def u_solution_func(out):
...
     train_dataloader_cfg = {
         "dataset": "IterableNamedArrayDataset",
-        "iters_per_epoch": ITERS_PER_EPOCH,
+        "iters_per_epoch": cfg.TRAIN.iters_per_epoch,
     }
 
-    NPOINT_INTERIOR = 99**2
-    NPOINT_BC = 400
+    NPOINT_INTERIOR = cfg.NPOINT_INTERIOR
+    NPOINT_BC = cfg.NPOINT_BC
```
Comment (HydrogenSulfate) [inline examples/laplace/laplace2d.py:54]: 这里两行可以删除，

File: examples/laplace/laplace2d.py [hunk 3]
```diff
@@ -50,11 +47,11 @@ def u_solution_func(out):
...
     NPOINT_TOTAL = NPOINT_INTERIOR + NPOINT_BC
```
Comment (HydrogenSulfate) [inline examples/laplace/laplace2d.py:55]: `NPOINT_TOTAL = cfg.NPOINT_INTERIOR + cfg.NPOINT_BC`

File: examples/laplace/laplace2d.py
Comment (HydrogenSulfate) [inline examples/laplace/laplace2d.py:1]: 1. 构造的几何的对角线坐标可以放YAML里：cfg.DIAGONAL_COORD
2. 优化器使用的学习率也放到YAML：cfg.TRAIN.learning_rate

File: examples/laplace/laplace2d.py [hunk 4]
```diff
@@ -134,18 +131,85 @@ def u_solution_func(out):
     # visualize prediction after finished training
     solver.visualize()
 
-    # directly evaluate model from pretrained_model_path(optional)
-    logger.init_logger("ppsci", f"{OUTPUT_DIR}/eval.log", "info")
+def evaluate(cfg: DictConfig):
+    # set random seed for reproducibility
+    ppsci.utils.misc.set_random_seed(cfg.seed)
+
```
Comment (HydrogenSulfate) [inline examples/laplace/laplace2d.py:137]: 删除137行

File: examples/laplace/laplace2d.py [hunk 5]
```diff
@@ -134,18 +131,85 @@ def u_solution_func(out):
...
+    # # set output directory
```
Comment (HydrogenSulfate) [inline examples/laplace/laplace2d.py:138]: 删除多余井号#

File: examples/laplace/laplace2d.py [hunk 6]
```diff
@@ -134,18 +131,85 @@ def u_solution_func(out):
...
+    logger.init_logger("ppsci", osp.join(cfg.output_dir, "train.log"), "info")
```
Comment (HydrogenSulfate) [inline examples/laplace/laplace2d.py:139]: `eval.log`

File: examples/laplace/laplace2d.py [hunk 7]
```diff
@@ -134,18 +131,85 @@ def u_solution_func(out):
...
+    # compute ground truth function
+    def u_solution_func(out):
+        """compute ground truth for u as label data"""
+        x, y = out["x"], out["y"]
+        return np.cos(x) * np.cosh(y)
+
+
+    NPOINT_INTERIOR = cfg.NPOINT_INTERIOR
+    NPOINT_BC = cfg.NPOINT_BC
+    NPOINT_TOTAL = NPOINT_INTERIOR + NPOINT_BC
```
Comment (HydrogenSulfate) [inline examples/laplace/laplace2d.py:159]: 修改同上

File: examples/laplace/laplace2d.py [hunk 8]
```diff
@@ -134,18 +131,85 @@ def u_solution_func(out):
...
+    if cfg.mode == "train":
+        train(cfg)
+    elif cfg.mode == "eval":
+        evaluate(cfg)
+    else:
+        raise ValueError(f"cfg.mode should in ['train', 'eval'], but got '{cfg.mode}'")
+
+
+if __name__ == "__main__":
+    main()
```
Comment (HydrogenSulfate) [inline examples/laplace/laplace2d.py:215]: 文件末尾换行

**Review by HydrogenSulfate — CHANGES_REQUESTED** [review]

File: docs/zh/examples/laplace2d.md [hunk 9]
```diff
@@ -2,6 +2,13 @@
 
 <a href="https://aistudio.baidu.com/aistudio/projectdetail/6169897?sUid=455441&shared=1&ts=1684122038217" class="md-button md-button--primary" style>AI Studio快速体验</a>
 
+=== "模型训练命令"
+
+    ``` sh
+    tar -xvf laplace2d_dataset.tar
```
Comment (HydrogenSulfate) [inline docs/zh/examples/laplace2d.md:8]: 删除`tar -xvf laplace2d_dataset.tar`

**Review by HydrogenSulfate — CHANGES_REQUESTED** [review]

File: examples/laplace/laplace2d.py [hunk 10]
```diff
@@ -134,18 +129,81 @@ def u_solution_func(out):
...
         validator=validator,
         visualizer=visualizer,
-        pretrained_model_path=f"{OUTPUT_DIR}/checkpoints/latest",
+        pretrained_model_path=cfg.EVAL.pretrained_model_path,
     )
     solver.eval()
-    # visualize prediction from pretrained_model_path(optional)
+    # visualize prediction after finished training
     solver.visualize()
+
```
Comment (HydrogenSulfate) [inline examples/laplace/laplace2d.py:197]: 函数和函数之间空两行

File: examples/laplace/laplace2d.py [hunk 11]
```diff
@@ -134,18 +129,81 @@ def u_solution_func(out):
     # visualize prediction after finished training
     solver.visualize()
```
Comment (HydrogenSulfate) [inline examples/laplace/laplace2d.py:131]: tar -xvf laplace2d_dataset.tar

**Review by HydrogenSulfate — CHANGES_REQUESTED** [review]

**Review by HydrogenSulfate — APPROVED** [review]
LGTM

**Comment by luotao1** [comment]
hi, @xusuyong 
- 非常感谢你对飞桨框架的贡献，我们正在运营一个PFCC组织，会通过定期分享技术知识与发布开发者主导任务的形式持续为飞桨框架做贡献，详情可见 https://github.com/luotao1 主页说明。
- 如果你对PFCC有兴趣，请发送邮件至 [ext_paddle_oss@baidu.com](mailto:ext_paddle_oss@baidu.com)，我们会邀请你加入~
